widgets/qt_pisces_main_window.py

```diff
@@ -16,80 +17,126 @@ def __init__(self, *args, **kwargs):
         self.top = 10
         self.width = 800
         self.height = 500
+        self.button_state = False
+
         self.initUI()
 
         self.show()  # not shown by default
 
     def initUI(self):  # Window properties are set in the initUI() method
+
+        @pyqtSlot()
+        def toggleState():
```

Comment:
this method is a closure right now, not a member method `MainWidgetWindow` class. Could you make it a member methed like `initUI()`?